### ISSUE TITLE
Regression with `-s` in 1.5 when input does not end with newline

### ISSUE THREAD

**Issue by emillon** [jqlang/jq#1170]
Hi,

I noticed a regression between 1.4 and 1.5 in the way `jq` handles end of file with `-s`:
### jq 1.4

```
% echo '{}{}{}{}' | jq -s ''  
[
  {},
  {},
  {},
  {}
]
% echo -n '{}{}{}{}' | jq -s ''
[
  {},
  {},
  {},
  {}
]
```
### jq 1.5

```
% echo '{}{}{}{}' | jq -s '' 
[
  {},
  {},
  {},
  {}
]
% echo -n '{}{}{}{}' | jq -s ''
[
  {}
]
{}
{}
{}
```

(as of now, `master` has the same problem as 1.5) I bisected the problem to 5bfb9781f7e5ba2ad99b71150bccc559d53b928f (Add Streaming parser)
I would expect the 1.4 behavior to be the correct one, since the JSON value is closed by `}` and not the newline character.

I'm unfortunately not familiar enough with the codebase to provide a patch but I'm happy to assist.

Thanks!

**Comment by pkoppstein** [comment]
This is to confirm that "jq -s" (in 1.5 and subsequently) is broken.  The relevant test cases can be simplified to:

    echo -n '{}{}' | jq -s .

and

    echo -n '{}{}' | jq -s length

**Comment by emillon** [comment]
Right. I previously used `length` which output 4 or `1 0 0 0` but subsequently edited my message to show how it parses the array.

**Comment by itchyny** [comment]
This looks like the same issue of #1534 and already fixed in f06deb828a318536b85d68280d429c3a70b21259, and released in [1.6](https://github.com/jqlang/jq/releases/tag/jq-1.6) in Nov 2018, so closing this issue.
